scipy/optimize/_remove_redundancy.py

```diff
@@ -230,6 +231,37 @@ def _remove_redundancy_dense(A, rhs):
     return A_orig[keep, :], rhs[keep], status, message
 
 
+class LUSolver:
+    def __init__(self, m):
+        I = eye(m, format='csc')
+        self.fl = splu(I, permc_spec="NATURAL")
+        self.fu = splu(I, permc_spec="NATURAL")
+        self.L = eye(m, format='csc')
+        self.U = eye(m, format='csc')
+
+    def solve(self, b, trans=False):
+        if trans:
+            y = self.fu.solve(b, 'T')
+            x = self.fl.solve(y, 'T')
+        else:
+            y = self.fl.solve(b)
+            x = self.fu.solve(y)
+        return x
+
+    def update(self, v, j):
+        L = self.L
+        U = self.U
+        m, n = L.shape
+
+        u = self.fl.solve(v.toarray())
+        U[:j+1, j] = u[:j+1]
+        l = u[j+1:]
+        piv = U[j, j]
+        L[j+1:, j] += (l/piv)
+        self.fl = splu(L, permc_spec="NATURAL", diag_pivot_thresh=0)
```

Comment:
This is quite fast because the matrix is already triangular. The options attempt to ensure that SuperLU doesn't do any row or column permutations. Ideally, after this operation `self.fl.L` will be `L` and `self.fl.U` will be the identity. Again, this is just a hack way of controlling the `L` and `U` attributes of a SuperLU object because I can't modify them directly.

Reply:
_Update: this *should* be quite fast, but it is the bottleneck._